modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -691,36 +691,58 @@ private long streamQuery0(String qry, String schemaName, IgniteDataStreamer stre
         try {
             UpdatePlan plan = dml.plan();
 
-            List<List<?>> planRows = plan.createRows(args != null ? args : X.EMPTY_OBJECT_ARRAY);
+            Object[] params = args != null ? args : X.EMPTY_OBJECT_ARRAY;
+
+            Iterator<List<?>> iter0;
+
+            if (!F.isEmpty(plan.selectQuery())) {
+                SqlFieldsQuery selectQry = new SqlFieldsQuery(plan.selectQuery())
+                    .setArgs(params)
+                    .setLocal(true);
```

Comment:
Why this query is always `true`?

Reply:
That behavior was implemented before was dropped at commit 6c9a17b4 (pr #6096 by vozerov), so I decided to return it with minimal differences.

Reply:
Why do you think that changes in 6c9a17b is wrong?

Reply:
Not all but condition removed from line 816 and assertion added to line 814.
That assertion was also removed at 1619dd56 (pr #6112).
Assume that adding of assertion is right, then it's logical to expect that we should meet the changes which fix that assertion failures happening when we use expressions at streaming inserts. But all further changes keep the bug described above till 1619dd56. After 1619dd56 we became to meet the bug described in our issue instead of assertion error.